app/static/favorite_quotes.js

```diff
@@ -0,0 +1,49 @@
+// Adding event listeners
+let hearts = Array.from(document.getElementsByClassName("heart"));
+window.addEventListener("load", function () {
+  hearts.forEach((heart) => {
+    if (heart) {
+      heart.addEventListener("click", onHeartClick);
+    }
+  });
+});
+
+const FULL_HEART = "../media/full_heart.png";
+const EMPTY_HEART = "../media/empty_heart.png";
+
+/**
+ * @summary This function is a handler for the event of heart-click.
+ * Whenever a user clicks on a heart icon, in case of empty heart:
+ * saves quote in favorites, as well as changing
+ * the heart icon from empty to full.
+ * In case of full heart:
+ * Removes it and switch back to empty heart icon.
+ * Uses the save_or_remove_quote function to handle db operations.
+ */
+function onHeartClick() {
+  const quote = this.parentNode.previousElementSibling.innerText;
```

Comment:
Prefer transferring the quote id, much more efficient and will save you in case of not unique quotes